### PR TITLE
Checkbox remote nomenclature documentation upgrade

### PR DESCRIPTION
**Pull request by pieqq**
## Description

After deciding on the new nomenclature (Testbed and Controller), the documentation needed to be updated.

This PR

1. introduces the new nomenclature in the Checkbox documentation
1. upgrades the github workflow to be able to run the woke command as expected
1. upgrades the glossary in the documentation
1. revamps the explanation about how the Checkbox Agent works 
1. rewrites the explanation about the interrupt screen provided by Checkbox remote, as done by @Hook25 in #540 
1. cleans up bad restructuredtext formatting

Part 4. requires some attention, because I'm not entirely sure about Checkbox internals with regards to snap/daemon/service and how each of these interact with each others. I've removed what I thought was not necessary, but I could be wrong.

**Click [here](https://canonical-checkbox--562.com.readthedocs.build/en/562/) to see the changes in action!**

I've split my work into different commits for easier review.

## Resolved issues

Fixes CHECKBOX-658

## Tests

Checked that pages looked OK during rewriting using `make run`, and checked inclusive language, spelling and links with

```
make woke && make spelling && make linkcheck
```

### CODE REVIEW

File: docs/how-to/agent-service.rst [hunk 1]
```diff
@@ -0,0 +1,72 @@
+Checkbox Agent Service
+^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (kissiel) [inline docs/how-to/agent-service.rst:2]: 2 extra chars

File: docs/how-to/agent-service.rst [hunk 2]
```diff
@@ -0,0 +1,72 @@
+Checkbox Agent Service
```
Comment (kissiel) [inline docs/how-to/agent-service.rst:1]: I think I'd add systemd before service so it doesn't sound weird :D

File: docs/explanation/configs.rst [hunk 3]
```diff
@@ -29,30 +35,30 @@ value resolution is as follows:
 Configs with Checkbox Remote
 ============================
 
-When the Checkbox Slave starts, it looks for config files in the same places
-that local Checkbox session would look (on the Slave side).
-If the Master uses a Launcher, then the values from that Launcher take
-precedence over the values from configs on the Slave side.
+When the :term:`Checkbox Agent` starts, it looks for config files in the same
+places that local Checkbox session would look (on the :term:`DUT` side). If
```
Comment (kissiel) [inline docs/explanation/configs.rst:39]: later on you use the phrase "run on the agent". Any reason why it's DUT here?

**Review by kissiel — COMMENTED** [review]
one small mistake, some comments and a question below

File: docs/how-to/agent-service.rst [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+.. code-block:: bash
+
+    $ sudo systemctl restart snap.checkbox.agent.service
+
+Stopping the service
+====================
+
+In a rare case where you want to have multiple Checkbox snaps installed on the
+system, it's necessary to disable all but one services.
```
[the remark this answers is not in the document]
Comment (pieqq) [reply docs/how-to/agent-service.rst]: Fixed in [c0681c6](https://github.com/canonical/checkbox/pull/562/commits/c0681c6b3ef6bc1d79d3a05f4269e64a65800ec6)

File: docs/how-to/agent-service.rst (hunk 1 above)
Comment (pieqq) [reply docs/how-to/agent-service.rst]: [c0681c6](https://github.com/canonical/checkbox/pull/562/commits/c0681c6b3ef6bc1d79d3a05f4269e64a65800ec6)

File: docs/how-to/agent-service.rst (hunk 2 above)
Comment (pieqq) [reply docs/how-to/agent-service.rst]: Fixed in [c0681c6](https://github.com/canonical/checkbox/pull/562/commits/c0681c6b3ef6bc1d79d3a05f4269e64a65800ec6)

File: docs/explanation/configs.rst (hunk 3 above)
Comment (pieqq) [reply docs/explanation/configs.rst]: I wanted to let users know Agents and DUT are interchangeable names, but I've replaced it with Agent as you suggest.

Fixed in [c0681c6](https://github.com/canonical/checkbox/pull/562/commits/c0681c6b3ef6bc1d79d3a05f4269e64a65800ec6)

**Review by yphus — APPROVED** [review]
Very good job with the new nomenclature, +1

File: docs/explanation/remote.rst [hunk 5]
```diff
@@ -14,82 +14,96 @@ This is especially useful on headless devices.
 Comparison with SSH
 ===================
 
-It's easy to lose SSH connection with the DUT, and if the device doesn't offer
-screen-like functionality then the Checkbox session has to be started over.
+It's easy to lose SSH connection with the :term:`Agent`, and if the device
```
Comment (kissiel) [inline docs/explanation/remote.rst:17]: The original text was about "losing" local checkbox session. This doesn't translate directly into Agent.

**Review by kissiel — CHANGES_REQUESTED** [review]
One thing I think needs changing.

**Review by kissiel — APPROVED** [review]
TYVM!
+1

**Pull request merged**